src/components/checkbox/README.md

```diff
@@ -1 +1,144 @@
+
+## When to consider something else
+
+When you need to present users with a list of mutually exclusive options and they can select only one option, consider using a list of [radio buttons](#) or a [select input](#).
+
+## Adding the markup
+
+The checkbox component is made up of a `form` element wrapping a `fieldset`. Inside the fieldset, there is a `legend` that is set to screenreader only using the class `rvt-sr-only`. Following the legend, there's a `ul` with the class `rvt-inline-list` (this sets the checkboxes to the default, inline list layout), and each list item contains a checkbox option: `input[type="checkbox"]` and `label`. The `id` must match with each `input[type="checkbox"]` and `label` pair, and the `name` attribute assigned to each of the `input` elements is the same for each `form`.
+
+```
+<form>
+    <fieldset>
+        <legend class="rvt-sr-only">Checkboxes inline</legend>
+        <ul class="rvt-inline-list">
+            <li>
+                <input type="checkbox" name="checkbox-demo" id="checkbox-1">
+                <label for="checkbox-1">Option one</label>
+            </li>
+            <li>
+                <input type="checkbox" name="checkbox-demo" id="checkbox-2">
+                <label for="checkbox-2">Option two</label>
+            </li>
+        </ul>
+    </fieldset>
+</form>
+```
+
+### Checkbox list
+
+To output the checkboxes similar to the vertical orientation of a `ul`, use the `rvt-plain-list` class on the `ul`.
+
+```
+<form>
+    <fieldset>
+        <legend class="sr-only">Checkbox list</legend>
+        <ul class="rvt-plain-list">
+            <li>
+                <input type="checkbox" name="checkbox-demo" id="checkbox-3">
+                <label for="checkbox-3" class="rvt-m-right-sm">Option one</label>
+            </li>
+            <li>
+                <input type="checkbox" name="checkbox-demo" id="checkbox-4">
+                <label for="checkbox-4">Option two</label>
+            </li>
+            <li>
+                <input type="checkbox" name="checkbox-demo" id="checkbox-4-disable" disabled>
+                <label for="checkbox-4-disable">Option three disabled</label>
+            </li>
+            <li>
+                <input type="checkbox" name="checkbox-demo" id="checkbox-5" disabled checked>
+                <label for="checkbox-5">Option four checked and disabled</label>
+            </li>
+        </ul>
+    </fieldset>
+</form>
+```
+
+### Indeterminate State
+
+It can be used to indicate a partially checked state. A typical use case would be nested checkboxes/selections like the following example. [This article](#) explains the indeterminate state in greater detail. **You will need to initialize the indeterminate JavaScript in order to use this functionality.** [UPDATE LATER WITH INSTRUCTIONS FOR HOW TO DO THIS]
```

Comment:
```suggestion
Indeterminate state is a built-in checkbox property that allows you to indicate a partially checked state. A typical use case would be nested checkboxes/selections like the following example. [This article](#) explains the indeterminate state in greater detail. **You will need to initialize the indeterminate JavaScript in order to use this functionality.**

To initialize the JavaScript for this, you will need to retrieve the element and set the indeterminate state to `true`.

```
// Initializing indeterminate state

```
<script>
const checkbox = document.querySelector('#checkbox-indeterminate');
    checkbox.indeterminate = true;
</script>
```

Reply:
The code suggestion logic has a bit of an issue with items wrapped in backticks within the suggestion, so this is a little messed up. Everything below the visual suggestion should be wrapped in backticks.